### PR TITLE
fix opensuse rpm

### LINKED ISSUE #19944: Issues with published RPMs for SuSE 13.2
There seems to be some problems with the packages generated for SuSE 13.2 and hosted on yum.dockerproject.org. Here are the commands I've been testing with:

Start a SuSE container:
```bash
docker run --rm -ti --privileged opensuse:13.2 bash
```

Install docker as follows:
```
zypper --non-interactive install curl ca-certificates*
zypper -n ar -f obs://Virtualization:containers/SLE_12 Virtualization:containers
# Import Docker's key and the SuSE key
rpm --import 'https://sks-keyservers.net/pks/lookup?op=get&search=0x58118E89F3A912897C070ADBF76221572C52609D'
rpm --import 'https://sks-keyservers.net/pks/lookup?op=get&search=0x55A0B34D49501BB7CA474F5AA193FBB572174FC2'
curl -O https://yum.dockerproject.org/repo/main/opensuse/13.2/Packages/docker-engine-1.9.1-1.x86_64.rpm
zypper install docker-engine-1.9.1-1.x86_64.rpm
```
this will prompt you to:
1. ignore missing `libcgroup` dependency (choose option 2)
2. accept installation (choose option y)
Not sure where the libcgroup package is supposed to come from.

After install, note the output similar to:
```
Additional rpm output:
/var/tmp/rpm-tmp.wLf4tV: line 1: fg: no job control
```
This indicates that one of the RPM scripts failed; in this case I suspect a missing macro.

After all of that:
```
docker --version
Docker version 1.9.1, build %{_gitcommit}
```
Note the incorrect build identifier "%{_gitcommit}".

I'm not sure how to best debug further. 

/cc @jfrazelle @mavenugo

### PR DESCRIPTION
**Pull request by jessfraz**
closes #19944

### CODE REVIEW

**Comment by jfrazelle** [comment]
ping @mikedougherty

File: contrib/builder/rpm/amd64/generate.sh [hunk 1]
```diff
@@ -106,8 +106,12 @@ for version in "${versions[@]}"; do
 
 	case "$from" in
 		opensuse:*)
-			packages=( "${packages[@]/btrfs-progs-devel/libbtrfs-devel}" )
+			packages=( "${packages[@]/btrfs-progs-devel/libbtrfs-devel }" )
```
Comment (tianon) [inline contrib/builder/rpm/amd64/generate.sh]: lol

File: contrib/builder/rpm/amd64/generate.sh [hunk 2]
```diff
@@ -106,8 +106,12 @@ for version in "${versions[@]}"; do
...
 			packages=( "${packages[@]/pkgconfig/pkg-config}" )
+			if [[ "$from" == "opensuse:13."* ]]; then
+				packages+=systemd-rpm-macros
```
Comment (tianon) [inline contrib/builder/rpm/amd64/generate.sh]: This should be `+=( ... )` right? (so it stays an array)

Comment (jfrazelle) [inline contrib/builder/rpm/amd64/generate.sh]: ya how is that working

On Wed, Feb 3, 2016 at 10:30 AM, Tianon Gravi <notifications@github.com>
wrote:

> In contrib/builder/rpm/amd64/generate.sh
> <https://github.com/docker/docker/pull/19972#discussion_r51764526>:
>
> >  			packages=( "${packages[@]/pkgconfig/pkg-config}" )
> > +			if [[ "$from" == "opensuse:13."* ]]; then
> > +				packages+=systemd-rpm-macros
>
> This should be +=( ... ) right? (so it stays an array)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/19972/files#r51764526>.
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

File: contrib/builder/rpm/amd64/generate.sh (hunk 1 above)
Comment (jfrazelle) [inline contrib/builder/rpm/amd64/generate.sh]: rebase whoes

On Wed, Feb 3, 2016 at 10:30 AM, Tianon Gravi <notifications@github.com>
wrote:

> In contrib/builder/rpm/amd64/generate.sh
> <https://github.com/docker/docker/pull/19972#discussion_r51764485>:
>
> > @@ -106,8 +106,12 @@ for version in "${versions[@]}"; do
> >
> >  	case "$from" in
> >  		opensuse:*)
> > -			packages=( "${packages[@]/btrfs-progs-devel/libbtrfs-devel}" )
> > +			packages=( "${packages[@]/btrfs-progs-devel/libbtrfs-devel }" )
>
> lol
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/19972/files#r51764485>.
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by tiborvass** [comment]
LGTM

**Comment by mikedougherty** [comment]
just tested, works for me. thanks!

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**